pest/src/parser_state.rs

```diff
@@ -1041,3 +1058,21 @@ impl<'i, R: RuleType> ParserState<'i, R> {
         self
     }
 }
+
+fn constrain_idxs(start_: i32, end_: Option<i32>, len: usize) -> Option<std::ops::Range<usize>> {
```

Comment:
current behavior: if a negative index is <0, fail to match. if `start` > `end` however, I consider this an empty slice of the stack and therefore matching.

TODO: also fail fast if `start` or `end` are >`len`. document all this.

Reply:
done